pig/src/main/java/org/jboss/pnc/bacon/pig/impl/repo/RepoManager.java

```diff
@@ -334,6 +349,133 @@ public RepositoryData generate() {
+                throw new IllegalStateException("Failed to resolve " + bom);
+            }
+            // Must point to a text file which contains list of format "groupId:artifactId:version"
+            String extensionsListUrl = params.get("extensionsListUrl");
+
+            List<Artifact> extensionRtArtifactList = parseExtensionsArtifactList(extensionsListUrl);
+            for (Artifact extensionRtArtifact : extensionRtArtifactList) {
+                // this will resolve all the runtime dependencies and as a consequence populate the local Maven repo
+                // specified in the user settings.xml
+                DependencyResult result = mvnResolver.resolveManagedDependencies(
+                        extensionRtArtifact, // runtime extension artifact
+                        Collections.emptyList(), // enforced direct dependencies, ignore this
+                        bomConstraints, // version constraints from the BOM
+                        Collections.emptyList(), // extra maven repos, ignore this
+                        "test",
+                        "provided" // dependency scopes that should be ignored
+                );
+
+            }
+            return repackage(sourceDir);
+        } catch (Exception bme) {
+            bme.printStackTrace();
+            return null;
+        }
+    }
+
+    /**
+     *
+     * @param urlString url to extensionList file. File must contains the extensions in format of
+     *                  groupId:artifactId:version
+     * @return List of Artifacts
+     * @throws Exception
+     */
+    private List<Artifact> parseExtensionsArtifactList(String urlString) throws Exception {
+        try {
+
+            URL url = new URL(urlString);
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            BufferedReader br = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+            ArrayList<Artifact> list = new ArrayList<>();
+            String buffer = "";
+            while (buffer != null) {
+                buffer = br.readLine();
+                if (buffer == null) {
+                    break;
+                }
+                String[] parts = buffer.split(":");
+                if (parts.length < 2 || parts.length > 3) {
+                    throw new Exception("Extension text file is not properly formatted");
+                }
+
+                Artifact extensionRtArtifact = new DefaultArtifact(parts[0], parts[1], null, "jar", parts[2]);
+                list.add(extensionRtArtifact);
+            }
+            return list;
+        } catch (MalformedURLException mfe) {
+            throw new MalformedURLException("url is not proper " + urlString);
+        } catch (IOException e) {
+            throw new IOException("Unable to do IO operation. Url: " + urlString);
+        }
+    }
+
+    private String getRedhatVersion(org.jboss.da.model.rest.GAV gav, boolean isTempBuild) {
```

Comment:
why do we need this? Can't we take the version from PNC build?

Reply:
I am not aware of `PncBuild` ? is that the one used in `GENERATE` strategy ?

Reply:
if you look around the code, there are several cases where we supply `groupId`, `artifactId` and name of a build and the version of the artifact is deduced from that

Reply:
I have used pnc Build this time to get version